### PR TITLE
Update to changes to construct_ssa

### PR DESCRIPTION
**Pull request by oxinabox**
This is an update to make things work after https://github.com/JuliaLang/julia/pull/50943

Right now however it seems to makes tests hang

### CODE REVIEW

File: src/stage1/recurse.jl [hunk 1]
```diff
@@ -276,7 +276,13 @@ function optic_transform!(ci, mi, nargs, N)
     stmts = VERSION < v"1.11.0-DEV.258" ? ir.stmts.inst : ir.stmts.stmt
```
Comment (topolarity) [inline src/stage1/recurse.jl:276]: FWIW in the typical pipeline `convert_to_ircode` is used to convert CodeInfo-style statements to IRCode, but that's skipped here.

Most of the function is a no-op if you have no (informative) results from inference but there is some normalization (e.g. `strip_trailing_junk!`) that might be needed here for certain IR.

None of that is new in this PR though, so we can worry about it later.

**Comment by oxinabox** [comment]
@topolarity any ideas why this would make it hang?
Seen both locally and in CI

**Comment by topolarity** [comment]
Not sure, but here's a reduced MWE at least:
```julia
using Diffractor

function times_three_while(x)
    z = x
    i = 3
    while i > 1
        z += x
        i -= 1
    end
    z
end

let var"'" = Diffractor.PrimeDerivativeBack
    println(times_three_while'(1.0) == 3.0)
end
```

The IR that Diffractor generates is identical on latest Julia versus `2cee483bce` ([ir.txt](https://github.com/JuliaDiff/Diffractor.jl/files/13025593/ir.txt)), but inference appears to get stuck in a loop doing semi-concrete eval of `>` and `-`:
```
semi-concrete eval: (::Diffractor.∂⃖{1})(typeof(Base.:(-)), Int64, Int64) from (::Diffractor.∂⃖{N})(T, Any...) where {T, N}
semi-concrete eval: (::Diffractor.∂⃖{1})(typeof(Base.:(>)), Int64, Int64) from (::Diffractor.∂⃖{N})(T, Any...) where {T, N}
semi-concrete eval: (::Diffractor.∂⃖{1})(typeof(Base.:(-)), Int64, Int64) from (::Diffractor.∂⃖{N})(T, Any...) where {T, N}
semi-concrete eval: (::Diffractor.∂⃖{1})(typeof(Base.:(>)), Int64, Int64) from (::Diffractor.∂⃖{N})(T, Any...) where {T, N}
... (repeats infinitely)
```

**Comment by oxinabox** [comment]
I guess since the IR is identical, we could consider merging this.

**Comment by aviatesk** [comment]
Going to merge as the 1.10 CI passes.

**Pull request merged**